### ISSUE TITLE
Initial connection time

### ISSUE THREAD

**Issue by TecbeckMacBook** [mongo-dart/mongo_dart#328]
I have used this package in my flutter app and connected my db with the following two lines:
```
var db = await Db.create("MY_URL");
await db.open();
```
These two lines are taking 10-15 seconds on average to execute which is huge time. User waits at the splash screen for this much time because I have to show data after fetching from mongo db on the homepage of my app. What should I do to make the initial connection faster?

**Comment by giorgiofran** [comment]
Where are you connecting?

**Comment by TecbeckMacBook** [comment]
> Where are you connecting?

Sorry, I didn't get your question.

**Comment by giorgiofran** [comment]
Are you connectin to some cloud service, like Atlas, or do you have a specific server.

**Comment by TecbeckMacBook** [comment]
> Are you connecting to some cloud service, like Atlas, or do you have a specific server.

I am using Atlas.

**Comment by giorgiofran** [comment]
Free Tier?

**Comment by TecbeckMacBook** [comment]
> Free Tier?

Yes.

**Comment by giorgiofran** [comment]
There were other person reporting this problem with Atlas, but I didn't get if it depends only on the Tier or if it is Atals itself.. If you try to connect locally or on a remote server created by you, the connection time is by far better.

**Comment by JohnF17** [comment]
I think we could confirm which has an "_issue_" by **running a few tests**, I for one, am currently running on the free tier atlas and i have **the same delay issue**, not only that but I also get a **random "No master connection" or "Connection lost" error** messages (_I'll manage to get the proper error messages next time_).  But **I did manage to get around that** by having an RX value to listen to those changes (in errors) and try to reconnect after a few seconds. **But going back to the main issue:**
- Check the connection type, is db.open(secure: true) faster, the same, just as bad or even worse? (_Checking this because when I get the **connection lost error,** it also tells me to try **without making tsl=true or secure: true**, **and I've done neither** so it should not disconnect but it happens_)
- Then since we know Current Free tier users have reported the supposed delay, we understand that **there is a delay on atlas + free**
- Now check if there are users that report delay issues on **atlas + paid**
- Then find other backers that **confirm it connects fast or barely slow when connecting to a remote server or locally** (_like you have confirmed_)

And then I think we'd have an understanding if its a problem with **Atlas itself, or the Free tier only**.

**Comment by DennisKragekjaer** [comment]
Yes same problem here, random Connection Exception, retry after some seconds and it works. But that is making me nervous.
